security/pkg/server/grpc/server_test.go

```diff
@@ -189,3 +188,63 @@ func TestShouldRefresh(t *testing.T) {
 		}
 	}
 }
+
+func TestRun(t *testing.T) {
+	testCases := map[string]struct {
+		ca                          *mockCA
+		hostname                    string
+		port                        int
+		expectedErr                 string
+		applyServerCertificateError string
+		expectedAuthenticatorsLen   int
+	}{
+		"Invalid listening port number": {
+			ca:          &mockCA{cert: csr},
+			port:        -1,
+			expectedErr: "cannot listen on port -1 (error: listen tcp: address -1: invalid port)",
+		},
+		"Random listening port number": {
+			ca:                        &mockCA{cert: csr},
+			hostname:                  "localhost",
+			port:                      0,
+			expectedErr:               "",
+			expectedAuthenticatorsLen: 2,
+			applyServerCertificateError: "tls: failed to find \"CERTIFICATE\" PEM block in certificate " +
+					"input after skipping PEM blocks of the following types: [CERTIFICATE REQUEST]",
+		},
+	}
+
+	for id, tc := range testCases {
+		server := New(tc.ca, tc.hostname, tc.port)
+		err := server.Run()
+		if len(tc.expectedErr) > 0 {
+			if err == nil {
+				t.Errorf("%s: Succeeded. Error expected: %v", id, err)
+			} else if err.Error() != tc.expectedErr {
+				t.Errorf("%s: incorrect error message: %s VS %s",
+					id, err.Error(), tc.expectedErr)
+			}
+			continue
+		} else if err != nil {
+			t.Fatalf("%s: Unexpected Error: %v", id, err)
```

Comment:
Errorf?

Reply:
if err is not nil, then server is nil. Tests below this line should be ignored.
I followed the way existing test case used. Fatalf can be changed with Errorf + continue.

Reply:
OK now I see. Could you replace all Errorf + continue to Fatalf? I think Fatalf is designed for the immediate fail purpose so more appropriate.

Reply:
Done